### PR TITLE
Merchant edits item#11

### LINKED ISSUE #11: User Story 56, Merchant edits an item
```
As a merchant
When I visit my items page
And I click the edit button or link next to any item
Then I am taken to a form similar to the 'new item' form
My URI route will be "/dashboard/items/15/edit" (if the item's ID was 15)
The form is re-populated with all of this item's information
I can change any information, but all of the rules for adding a new item still apply:
- name and description cannot be blank
- price cannot be less than $0.00
- inventory must be 0 or greater

When I submit the form
I am taken back to my items page
I see a flash message indicating my item is updated
I see the item's new information on the page, and it maintains its previous enabled/disabled state
If I left the image field blank, I see a placeholder image for the thumbnail
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #18

depends on #13

### LINKED ISSUE #149: Refactor: Add dashboard namespace or similar

### PR DESCRIPTION
**Pull request by asmolentzov**
Completes User Story 56, Merchant edits an item
## Main functionality:
- Adds the ability for a merchant to edit an item
- The merchant can edit any or all information for the item
- Shows a flash message upon success
- Shows error messages if information is entered incorrectly
- Standard restrictions still apply:
  - Name or description cannot be blank
  - Inventory must be greater than 0
  - Price must be greater than 0
- The image link can be left blank, and is then updated with the default image
- Added validations (and tests) to Item model for inventory and price being integers greater than or equal to 0
- Adds partial view for the item information form, so that it can be shared by both the new and edit pages

## Bonus functionality that became necessary for the partial form to work properly:
- Creates dashboard namespace in routes
- Updates controllers and views accordingly
- Cleans up some code where `if current_merchant?` logic is no longer necessary 
- Used failing tests to guide me on where to clean up code in switching to dashboard namespace 
- *** If there is anything you suspect might be affected by this, please check that in particular!

- All tests passing

Closes #11 
Closes #149

### CODE REVIEW

File: spec/models/item_spec.rb [hunk 1]
```diff
@@ -12,8 +12,8 @@
     it { should validate_presence_of(:image_link)}
     it { should validate_presence_of(:description)}
     it { should validate_inclusion_of(:enabled).in_array([true, false])}
-    it { should validate_inclusion_of(:inventory).in_range(1..1000000)}
-    it { should validate_presence_of(:current_price)}
+    it { should validate_numericality_of(:inventory).only_integer.is_greater_than_or_equal_to(0)}
```
Comment (danbriechle) [inline spec/models/item_spec.rb]: 👍
